contracts/Market.sol

```diff
@@ -927,7 +936,12 @@ contract Market is
         amount: unrestrictedSupplierFee
       });
     }
-    bytes memory data = abi.encode(recipient, certificateAmount);
+    bytes memory data = abi.encode(
+      recipient,
+      certificateAmount,
+      address(_purchasingToken),
```

Comment:
Sort of wonder if there's a way to detect this without passing it through (either by parsing out this information from transaction data since the token contract address and amount will be included in the originating transaction) or some other way (msg.sender? tx.origin?-- I think both of these are invalid for different reasons but that's the initial direction my mind is drawing me towards).

Reply:
Also, is it possible to instead pass a struct as the data argument somehow? I vaguely recall some solidity magic that let me do this before, but I'm not sure what was required to pull it off

Reply:
I think you may be mixing up our old approach of using ERC777 tokens where the token contract address would be present in the originating transaction but that is no longer the case with the permit mechanism